include/zephyr/bluetooth/audio/audio.h

```diff
@@ -371,20 +350,37 @@ struct bt_audio_codec_cfg {
 	uint16_t cid;
 	/** Codec Company Vendor ID */
 	uint16_t vid;
-#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT)
-	/** Codec Specific Configuration Data count */
-	size_t   data_count;
-	/** Codec Specific Configuration Data */
-	struct bt_audio_codec_data data[CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT];
-#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT */
-#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT)
-	/** Codec Specific Configuration Metadata count */
-	size_t   meta_count;
-	/** Codec Specific Configuration Metadata */
-	struct bt_audio_codec_data meta[CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT];
-#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT */
+#if CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE > 0
+	/** Codec Specific Capabilities Data count */
+	size_t data_len;
+	/** Codec Specific Capabilities Data */
+	uint8_t data[CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE];
+#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE > 0 */
+#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE)
```

Comment:
```suggestion
#if CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE > 0
```
Should this one be changed as well, just like you did in 353?